HW4_Gorbarenko/amino_analyzer.py

```diff
@@ -0,0 +1,196 @@
+
+    Returns:
+        tuple: A tuple containing the count of hydrophobic and hydrophilic amino acids, respectively.
+    """
+    hydrophobic_aa = ['A', 'V', 'L', 'I', 'P', 'F', 'W', 'M']
+    hydrophilic_aa = ['R', 'N', 'D', 'C', 'Q', 'E', 'G', 'H', 'K', 'S', 'T', 'Y']
+    
+    hydrophobic_count = 0
+    hydrophilic_count = 0
+    
+    seq = seq.upper()
+    
+    for aa in seq:
+        if aa in hydrophobic_aa:
+            hydrophobic_count += 1
+        elif aa in hydrophilic_aa:
+            hydrophilic_count += 1
+    
+    return hydrophobic_count, hydrophilic_count
+
+def peptide_cutter(sequence: str, enzyme: str = "trypsin") -> str:
+    """
+    This function identifies cleavage sites in a given peptide sequence using a specified enzyme.
+    
+    Args:
+        sequence (str): The input peptide sequence.
+        enzyme (str): The enzyme to be used for cleavage. Choose between "trypsin" and "chymotrypsin". Default is "trypsin".
+        
+    Returns:
+        str: A message indicating the number and positions of cleavage sites, or an error message if an invalid enzyme is provided.
+    """
+    cleavage_sites = []
+    if enzyme not in ("trypsin", "chymotrypsin"):
+        return "You have chosen an enzyme that is not provided. Please choose between trypsin and chymotrypsin."
+    
+    if enzyme == "trypsin":  # Trypsin cuts peptide chains mainly at the carboxyl side of the amino acids lysine or arginine.
+        for i in range(len(sequence)-1):
+            if sequence[i] in ['K', 'R', 'k', 'r'] and sequence[i+1] not in ['P','p']:
+                cleavage_sites.append(i+1)
+    
+    if enzyme == "chymotrypsin":  # Chymotrypsin preferentially cleaves at Trp, Tyr and Phe in position P1(high specificity) 
+        for i in range(len(sequence)-1):
+            if sequence[i] in ['W', 'Y', 'F', 'w', 'y', 'f'] and sequence[i+1] not in ['P','p']:
+                cleavage_sites.append(i+1)
+    
+    if cleavage_sites:
+        return f"Found {len(cleavage_sites)} {enzyme} cleavage sites at positions {', '.join(map(str, cleavage_sites))}"
+    else:
+        return f"No {enzyme} cleavage sites were found."
+
+
+def one_to_three_letter_code(sequence: str) -> str:
+    """
+    This function converts a protein sequence from one-letter amino acid code to three-letter code.
+    
+    Args:
+        sequence (str): The input protein sequence in one-letter code.
+        
+    Returns:
+        str: The converted protein sequence in three-letter code.
+    """
+    amino_acids = {
+        'A': 'Ala', 'C': 'Cys', 'D': 'Asp', 'E': 'Glu', 'F': 'Phe',
+        'G': 'Gly', 'H': 'His', 'I': 'Ile', 'K': 'Lys', 'L': 'Leu',
+        'M': 'Met', 'N': 'Asn', 'P': 'Pro', 'Q': 'Gln', 'R': 'Arg',
+        'S': 'Ser', 'T': 'Thr', 'V': 'Val', 'W': 'Trp', 'Y': 'Tyr'
+    }
+    
+    three_letter_code = [amino_acids.get(aa.upper()) for aa in sequence]
+    return ''.join(three_letter_code)
```

Comment:
Я бы лучше сделала '-'.join(three_letter_code), так запись более читаема (и, если мне не изменяет память, так трехбуквенные аминокислотные остатки и записывают). Например, Val-Ile-Leu-Glu (вместо ValIleLeuGlu)
```suggestion
    return '-'.join(three_letter_code)
```